backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IrToBitcode.kt

```diff
@@ -2626,13 +2627,20 @@ internal class CodeGeneratorVisitor(val context: Context, val lifetimes: Map<IrE
     private fun appendLlvmUsed(name: String, args: List<LLVMValueRef>) {
         if (args.isEmpty()) return
 
-        memScoped {
-            val argsCasted = args.map { it -> constPointer(it).bitcast(int8TypePtr) }
-            val llvmUsedGlobal =
+        val argsCasted = args.map { it -> constPointer(it).bitcast(int8TypePtr) }
+        val llvmUsedGlobal =
                 context.llvm.staticData.placeGlobalArray(name, int8TypePtr, argsCasted)
 
-            LLVMSetLinkage(llvmUsedGlobal.llvmGlobal, LLVMLinkage.LLVMAppendingLinkage)
-            LLVMSetSection(llvmUsedGlobal.llvmGlobal, "llvm.metadata")
+        LLVMSetLinkage(llvmUsedGlobal.llvmGlobal, LLVMLinkage.LLVMAppendingLinkage)
+        LLVMSetSection(llvmUsedGlobal.llvmGlobal, "llvm.metadata")
+    }
+
+    private fun appendDebugSelector() {
+        if (context.config.produce.isNativeBinary) {
+            val llvmDebugSelector =
+                    context.llvm.staticData.placeGlobal("KonanNeedDebugInfo",
+                            Int32(if (context.shouldContainDebugInfo()) 1 else 0))
+            LLVMSetLinkage(llvmDebugSelector.llvmGlobal, LLVMLinkage.LLVMExternalLinkage)
```

Comment:
Why not pass `isExternal = true` to `placeGlobal` instead?

Reply:
It only got `isExported` it seems.

Reply:
Yes, my bad, I have meant `isExported`.